src/main/java/com/fasterxml/jackson/databind/MapperFeature.java

```diff
@@ -541,15 +552,20 @@ public enum MapperFeature implements ConfigFeature
     ALLOW_EXPLICIT_PROPERTY_RENAMING(false),
 
     /**
-     * Feature that when enabled will allow getters with is-Prefix also for
-     * non-boolean return types; if disabled only methods that return
-     * {@code boolean} or {@code Boolean} qualify as "is getters".
+     * Feature that can be enabled to solve problem where an upper-case letter in
+     * the first 2 characters of Java field name (like {@code "IPhone"} or {@code "iPhone"})
+     * prevents match with property name derived from accessors (setter like
+     * {@code getIPhone()} becomes {@code "iphone"}).
+     * If enabled, additional checking is done with case-insensitive comparison (for
+     * cases of the first or second letter of Field name being upper-case) to merge
+     * accessors. If disabled, no special processing is done.
      * <p>
-     * Feature is disabled by default for backwards compatibility.
+     * Feature is disabled by default in 2.x for backwards-compatibility.
+     * It will be enabled by default in 3.0.
      *
-     * @since 2.14
+     * @since 2.20
      */
-    ALLOW_IS_GETTERS_FOR_NON_BOOLEAN(false),
+    FIX_FIELD_NAME_CASE_MISMATCH(false),
```

Comment:
My concern is...  Would the name imply we would fix any case difference issue?

Reply:
Not really. So maybe should still consider name to use...

In a way it is not necessarily limited to first two, but it only checks cases where Field name has first and/or second character upper case. Since those are ones where name-mangling can produce differences. But comparison itself consider whole implicit names.
So bit tricky to explain.

Alternatively could change logic itself to actually consider all cases of "lone" (unmatched) Fields, either with at least one upper-case letter, or possibly just any (since Getter/Setter name could have upper-case even without Field)